Content.Client/Damage/DamageVisualsSystem.cs

```diff
@@ -376,11 +376,16 @@ private void HandleDamage(AppearanceComponent component, DamageVisualsComponent
         if (damageVisComp.TrackAllDamage)
         {
             UpdateDamageVisuals(damageComponent, spriteComponent, damageVisComp);
+            return;
         }
-        else if (AppearanceSystem.TryGetData<DamageVisualizerGroupData>(component.Owner, DamageVisualizerKeys.DamageUpdateGroups, out var data, component))
+
+        if (!AppearanceSystem.TryGetData<DamageVisualizerGroupData>(uid, DamageVisualizerKeys.DamageUpdateGroups,
+                out var data, component))
         {
-            UpdateDamageVisuals(data.GroupList, damageComponent, spriteComponent, damageVisComp);
+            data = new DamageVisualizerGroupData(Comp<DamageableComponent>(uid).DamagePerGroup.Keys.ToList());
```

Comment:
This is the main bug being fixed - it now uses a default value if the appearance component has no data,